pkg/apis/submariner/v1alpha1/submariner_types.go

```diff
@@ -46,6 +46,7 @@ type SubmarinerSpec struct {
 	GlobalCIDR               string `json:"globalCIDR,omitempty"`
 	Namespace                string `json:"namespace"`
 	CableDriver              string `json:"cableDriver,omitempty"`
+	ServiceDiscoveryEnabled  bool   `json:"serviceDiscoveryEnabled,omitempty"`
```

Comment:
What is this for here?

Reply:
If this is enabled Submariner operator will create service discovery CR and that will in-turn deploy the lighthouse-agent.

Reply:
yeah I actually realized that when I looked at the controller code you added 😄